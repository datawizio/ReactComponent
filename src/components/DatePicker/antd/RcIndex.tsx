import Picker from "./RcPicker";
import { PickerProps } from "rc-picker/es/Picker";
import PickerPanel, { PickerPanelProps } from "rc-picker/es/PickerPanel";
import RangePicker from "./RcRangePicker";
import { RangePickerProps } from "rc-picker/es/RangePicker";

export {
  PickerPanel,
  RangePicker,
  PickerProps,
  PickerPanelProps,
  RangePickerProps
};

export default Picker;
