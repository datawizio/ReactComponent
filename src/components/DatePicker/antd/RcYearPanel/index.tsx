import * as React from "react";
import YearHeader from "rc-picker/lib/panels/YearPanel/YearHeader";
import YearBody, { YEAR_COL_COUNT } from "./YearBody";
import type { PanelSharedProps, PanelMode } from "rc-picker/lib/interface";
import { createKeyDownHandler } from "rc-picker/lib/utils/uiUtil";
import { calendarInfo } from "../../../../utils/calendar";

export type YearPanelProps<DateType> = {
  sourceMode: PanelMode;
} & PanelSharedProps<DateType>;

export const YEAR_DECADE_COUNT = 10;

function YearPanel<DateType>(props: YearPanelProps<DateType>) {
  const {
    prefixCls,
    operationRef,
    onViewDateChange,
    generateConfig,
    value,
    viewDate,
    sourceMode,
    onSelect,
    onPanelChange
  } = props;

  let panelPrefixCls = `${prefixCls}-year-panel`;
  //@ts-ignore
  if (generateConfig.type === "fiscal" || calendarInfo.startMonth !== 0) {
    panelPrefixCls += ` ${prefixCls}-year-panel-fiscal`;
  }

  // ======================= Keyboard =======================
  operationRef.current = {
    onKeyDown: event =>
      createKeyDownHandler(event, {
        onLeftRight: diff => {
          onSelect(generateConfig.addYear(value || viewDate, diff), "key");
        },
        onCtrlLeftRight: diff => {
          onSelect(
            generateConfig.addYear(value || viewDate, diff * YEAR_DECADE_COUNT),
            "key"
          );
        },
        onUpDown: diff => {
          onSelect(
            generateConfig.addYear(value || viewDate, diff * YEAR_COL_COUNT),
            "key"
          );
        },
        onEnter: () => {
          onPanelChange(
            sourceMode === "date" ? "date" : "month",
            value || viewDate
          );
        }
      })
  };

  // ==================== View Operation ====================
  const onDecadeChange = (diff: number) => {
    const newDate = generateConfig.addYear(viewDate, diff * 10);
    onViewDateChange(newDate);
    onPanelChange(null, newDate);
  };

  return (
    <div className={panelPrefixCls}>
      <YearHeader
        {...props}
        prefixCls={prefixCls}
        onPrevDecade={() => {
          onDecadeChange(-1);
        }}
        onNextDecade={() => {
          onDecadeChange(1);
        }}
        onDecadeClick={() => {
          onPanelChange("decade", viewDate);
        }}
      />
      <YearBody
        {...props}
        prefixCls={prefixCls}
        onSelect={date => {
          onPanelChange(sourceMode === "date" ? "date" : "month", date);
          onSelect(date, "mouse");
        }}
      />
    </div>
  );
}

export default YearPanel;
