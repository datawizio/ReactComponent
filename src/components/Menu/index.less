@import "~antd/es/menu/style/index";
@import "../../styles/variables";

@menu-bg: @bg-dark-color;
@menu-dark-bg: @bg-dark-color;

.ant-dropdown-menu-item.danger {
  color: @red-6;

  .ant-upload {
    color: @red-6;
  }
  &.ant-dropdown-menu-item-disabled {
    color: @ant-menu-item-disabled-color
    .ant-upload {
      color: @ant-menu-item-disabled-color;
    }
  }
}
