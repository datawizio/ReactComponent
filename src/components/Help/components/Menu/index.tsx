import * as React from "react";
import { useContext, useMemo } from "react";
import { Menu, Dropdown, Button } from "antd";
import { QuestionCircleOutlined } from "@ant-design/icons";
import ConfigContext from "../../../ConfigProvider/context";
import "./index.less";

export interface IHelpMenu {
  onTutorialClick: () => void;
  onSupportClick: () => void;
  onServiceUpdateClick: () => void;
}

const HelpMenu: React.FC<IHelpMenu> = ({ onTutorialClick, onSupportClick, onServiceUpdateClick }) => {
  const { translate } = useContext(ConfigContext);

  const menu = useMemo(() => {
    return (
      <Menu theme="light" className="help-menu-dropdown">
        <Menu.Item key="1" onClick={onTutorialClick}>
          {translate("READ_TUTORIAL")}
        </Menu.Item>
        <Menu.Item key="2" onClick={onSupportClick}>
          {translate("SUPPORT")}
        </Menu.Item>
        <Menu.Item key="3" onClick={onServiceUpdateClick}>
          {translate("SERVICE_UPDATE")}
        </Menu.Item>
      </Menu>
    );
  }, [onServiceUpdateClick, onSupportClick, onTutorialClick, translate]);

  return (
    <>
      <Dropdown overlay={menu} trigger={["click"]} placement="bottomRight">
        <Button type="link" className="help-icon" onClick={e => e.preventDefault()}>
          <QuestionCircleOutlined />
        </Button>
      </Dropdown>
    </>
  );
};

export default HelpMenu;