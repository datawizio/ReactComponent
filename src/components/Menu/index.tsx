import * as React from "react";

import { Menu as AntMenu } from "antd";
import { MenuProps as AntMenuProps } from "antd/lib/menu";

import "./index.less";

export interface MenuProps extends AntMenuProps {}

export type FCMenuProps = React.FC<MenuProps>;

export interface FCMenu extends FCMenuProps {
  Item: typeof AntMenu.Item;
  SubMenu: typeof AntMenu.SubMenu;
}

const Menu: FCMenu = props => {
  return <AntMenu {...props} />;
};

Menu.defaultProps = {
  theme: "dark"
};

Menu.Item = AntMenu.Item;
Menu.SubMenu = AntMenu.SubMenu;

export default Menu;
