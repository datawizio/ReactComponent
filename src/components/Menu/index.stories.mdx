import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Menu, { Item } from "./index";
import {
  PieChartOutlined,
  DesktopOutlined,
  ContainerOutlined
} from "@ant-design/icons";

<Meta title="Components/Menu/Menu" component={Menu} />

# Menu

Компонент `Menu`. Базовый компонент:
[**antd/menu**](https://ant.design/components/menu/).

## Basic

<Preview>
  <Story name="Basic">
    <div style={{ width: 256 }}>
      <Menu defaultSelectedKeys={["1"]} mode="inline">
        <Item key="1" icon={<PieChartOutlined />}>
          Активність користувачів
        </Item>
        <Item key="2" icon={<DesktopOutlined />}>
          Користувачі
        </Item>
        <Item key="3" icon={<ContainerOutlined />}>
          Ролі
        </Item>
        <Item key="4" icon={<ContainerOutlined />}>
          Історія активності
        </Item>
        <Item key="5" icon={<ContainerOutlined />}>
          Налаштування
        </Item>
        <Item key="6" icon={<ContainerOutlined />}>
          Доступні додатки
        </Item>
      </Menu>
    </div>
  </Story>
</Preview>

## Props

- [**antd/menu**](https://ant.design/components/menu/#API)

<Props components={{ Menu, Item }} />
