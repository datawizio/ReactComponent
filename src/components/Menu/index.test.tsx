import "jsdom-global/register";
import React from "react";
import { Menu as AntMenu } from "antd";
import { shallow } from "enzyme";

import Menu from "./index";

const mockProps = {
  Item: AntMenu.Item,
  SubMenu: AntMenu.SubMenu,
  ItemGroup: AntMenu.ItemGroup,
  Divider: AntMenu.Divider,
};

const setUp = (props?) => shallow(<Menu {...props} />);

describe("Menu component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Menu' rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
