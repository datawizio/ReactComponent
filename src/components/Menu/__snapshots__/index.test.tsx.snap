// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Menu' rendered correctly 1`] = `
<Menu
  Divider={[Function]}
  Item={[Function]}
  ItemGroup={[Function]}
  SubMenu={[Function]}
  theme="dark"
/>
`;
